centreon/src/Core/ServiceGroup/Infrastructure/Repository/DbReadServiceGroupRepository.php

```diff
@@ -390,7 +390,51 @@ private function retrieveServiceGroups(
             'alias' => 'sg.sg_alias',
             'name' => 'sg.sg_name',
             'is_activated' => 'sg.sg_activate',
+            'host_id' => 'h.host_id',
+            'host_group_id' => 'hg.hg_id',
+            'host_category_id' => 'hc.hc_id',
         ]);
+        if (mb_strpos($requestParameters?->getSearchAsString() ?? '', 'host_group_id')) {
+            $concatenator->appendJoins(
+                <<<'SQL'
+                    LEFT JOIN `:db`.acl_resources_sg_relations arsr
+                        ON sg.sg_id = arsr.sg_id
+                    LEFT JOIN `:db`.acl_resources res
+                            ON arsr.acl_res_id = res.acl_res_id
+                    LEFT JOIN `:db`.acl_resources_hg_relations arhgr
+                        ON res.acl_res_id = arhgr.acl_res_id
+                    LEFT JOIN `:db`.hostgroup hg
+                        ON arhgr.hg_hg_id = hg.hg_id
+                    SQL
+            );
+        } else if (mb_strpos($requestParameters?->getSearchAsString() ?? '', 'host_id')) {
+            $concatenator->appendJoins(
+                <<<'SQL'
+                    LEFT JOIN `:db`.acl_resources_sg_relations arsr
+                        ON sg.sg_id = arsr.sg_id
+                    LEFT JOIN `:db`.acl_resources res
+                            ON arsr.acl_res_id = res.acl_res_id
+                    LEFT JOIN `:db`.acl_resources_host_relations arhr
+                        ON res.acl_res_id = arhr.acl_res_id
+                    LEFT JOIN `:db`.host h
+                        ON arhr.host_host_id = h.host_id
+                    SQL
+            );
+        } else if (mb_strpos($requestParameters?->getSearchAsString() ?? '', 'host_category_id')) {
+            $concatenator->appendJoins(
+                <<<'SQL'
+                    LEFT JOIN `:db`.acl_resources_sg_relations arsr
+                        ON sg.sg_id = arsr.sg_id
+                    LEFT JOIN `:db`.acl_resources res
+                            ON arsr.acl_res_id = res.acl_res_id
+                    LEFT JOIN `:db`.acl_resources_hc_relations arhcr
+                        ON res.acl_res_id = arhcr.acl_res_id
+                    LEFT JOIN `:db`.hostcategories hc
+                        ON arhcr.hc_id = hc.hc_id
+                        AND hc.level IS NOT NULL
+                    SQL
```

Comment:
⚠️ This doesn't work
The servicegroups listed will be in the same ACLs Resource group as the host_group/host/host_category but it does'nt mean they are linked in terms of resource configuration
You need to join on the relation tables of servicegroup-service, service-host, host-hostgroup (for example)

But as in you other PRs on the same subject:
⚠️ If you are in an on-prem platform with non-admin user,
you have to filter hostgroups, host and hostcategories on the user's ACLs

I talked about it with @dmyios, you can see with him for more detailed explanations